src/function_space/shape_function_generator.jl

```diff
@@ -0,0 +1,411 @@
+# This file gathers shape function generation utils
+
+using Symbolics
+using LinearAlgebra
+using ForwardDiff
+
+@variables x[1:3]
+
+abstract type GeneratedFiniteElement <: AbstractFunctionSpaceType end
+struct LagrangeGenerated <: GeneratedFiniteElement end
+struct HermiteGenerated <: GeneratedFiniteElement end
+
+function FunctionSpace(::Val{:LagrangeGenerated}, degree::Integer)
+    FunctionSpace(LagrangeGenerated(), degree)
+end
+function FunctionSpace(::Val{:HermiteGenerated}, degree::Integer)
+    FunctionSpace(HermiteGenerated(), degree)
+end
+
+basis_functions_style(::FunctionSpace{<:LagrangeGenerated}) = NodalBasisFunctionsStyle()
+
+abstract type LinearForm end
+
+"""
+  Defines a point evaluation linear functional l(f) = f(x) (x is fixed)
+"""
+struct PointEvaluationForm <: LinearForm
+    x::Vector{Float64}
+end
+
+function (l::PointEvaluationForm)(f::Function)
+    return f(l.x)
+end
+
+"""
+  Defines a derivative point evaluation linear functional l(f) = f'(x) (x is fixed)
+"""
+struct PointDerivativeForm <: LinearForm
+    x::Vector{Float64}
+end
+
+function (l::PointDerivativeForm)(f::Function)
+    dfdx = x -> ForwardDiff.gradient(f, x)[1]
+    return dfdx(l.x)
+end
+
+"""
+  Returns the linear forms for continous Lagrange shape functions on a line => evaluation of f at nodes
+"""
+function Forms(::LagrangeGenerated, ::Line, deg)
+    forms = Array{LinearForm}(undef, deg + 1)
+    for i in 1:(deg + 1)
+        x = [-1.0 + (i - 1) * 2.0 / (deg)]
+        forms[i] = PointEvaluationForm(x)
+    end
+
+    return forms
```

Comment:
you can "simplify" the whole function with `forms = [PointEvaluationForm([-1. + (i-1)*2/deg]) for i in 1:(deg+1)]`